geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/conf/partition/TimePartition.scala

```diff
@@ -60,13 +60,18 @@ class TimePartition(metadata: GeoMesaMetadata[String], typeName: String, dtg: St
     f"${toBin(ZonedDateTime.ofInstant(toInstant(date), ZoneOffset.UTC))}%05d" // a short should fit into 5 digits
   }
 
-  override def partitions(filter: Filter): Seq[String] = {
+  override def partitions(filter: Filter): Option[Seq[String]] = {
```

Comment:
Ok, do you have a really good reason for using Option[Seq]?  Seems like this could be done with a Seq.  

That said, tell me you needed it, and I'll ship the PR:).

Reply:
yeah, I tried to show cover it in the scaladocs: https://github.com/locationtech/geomesa/blob/05d69812e866566b46880ecd3facd587d0017253/geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/conf/partition/TablePartition.scala#L33
Basically we need to differentiate between "there are no matching partitions" and "could not extract any partitions from the predicate"